R/htmlwidgets.R

```diff
@@ -226,7 +226,9 @@ widget_html <- function(name, package, id, style, class, inline = FALSE, ...){
 
   # call the custom function if we have one, otherwise create a div
   if (is.function(fn)) {
-    fn(id = id, style = style, class = class, ...)
+    renderTags(
```

Comment:
I don't know if this is the right fix. It might rather be to surround the results of the function in an `html-preserve` comment (which will also prevent pandoc from touching it). @jcheng5 What's the best thing here?